content/blog/vscode-esp-idf-v1.8.0/index.md

```diff
@@ -0,0 +1,114 @@
+
+{{< figure
+    default=true
+    src="img/debug-arch.webp"
+    alt=""
+    caption=""
+    >}}
+
+This intermediary is called the Debug Adapter (or DA for short) and the abstract protocol that is used between the DA and VS Code is the Debug Adapter Protocol (DAP for short). Since the Debug Adapter Protocol is independent from VS Code, it has its own [web site](https://microsoft.github.io/debug-adapter-protocol/) and many developers have implemented.
+
+## Eclipse CDT Debug Adapter
+
+Before we have implemented our own [ESP-IDF Debug Adapter](https://github.com/espressif/esp-debug-adapter) in Python. While this debug adapter works, our users have report many issues regarding configuration, responsiveness and runtime errors. In our search for solution we encountered that the Eclipse CDT Cloud team have published a [CDT-GDB-Adapter](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter) which is written in TypeScript and is using [NPM serialport package](https://www.npmjs.com/package/serialport) underneath. We added a few changes in order to make it work with our tools but it is mostly the same as the original.
+
+To use it in a ESP-IDF project `.vscode/launch.json`, you can add this configuration:
+
+```JSON
+{
+  "configurations": [
+    {
+      "type": "gdbtarget",
+      "request": "attach",
+      "name": "Eclipse CDT GDB Adapter"
+    }
+  ]
+}
+```
+
+After you build (**ESP-IDF: Build your Project**), flash (**ESP-IDF: Flash your Project**) you can go to menu `Run` and press `Start debugging`. You can check how to use the new debugger by following the [debug tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/debugging.md).
+
+{{< figure
+    default=true
+    src="img/debug-example.webp"
+    alt=""
+    caption=""
+    >}}
+
+## Advanced configuration of Eclipse CDT Debug Adapter
+
+There are many configurable options to customize the debug adapter if desired. You can view a deeper description in the [debugging documentation](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/DEBUGGING.md).
+
+## ESP-IDF version switcher
+
+Whenever you use the **ESP-IDF: Configure ESP-IDF Extension** to install ESP-IDF in your system, these installation are saved in our extension as global state IDF setup. You can now use the **ESP-IDF: Select Current ESP-IDF Version** to set in your current project (workspace folder in Visual Studio Code language) the ESP-IDF version you desire to use. If you want to delete these IDF setups from the extension global state you can use the **ESP-IDF: Clear Saved ESP-IDF Setups**.
+
+{{< figure
+    default=true
+    src="img/idf-version-list.webp"
+    alt=""
+    caption=""
+    >}}
+
+The current ESP-IDF version for currently open ESP-IDF project is shown in the status bar items like this:
+
+{{< figure
+    default=true
+    src="img/idf-version-status-bar.webp"
+    alt=""
+    caption=""
+    >}}
+
+## Add ESP-IDF: idf.py reconfigure task command
```

Comment:
If you choose to update this section title according to my suggestion above, then:

```suggestion
## Use code navigation without full project build
```